pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -614,11 +622,19 @@ def get_result(self, handle: ResultHandle, **kwargs: KwargTypes) -> BackendResul
                         sleep(10)
 
                 res = job.result(timeout=kwargs.get("timeout", None))
-            assert isinstance(res, SamplerResult)
-            for circ_index, (r, d) in enumerate(zip(res.quasi_dists, res.metadata)):
-                self._ibm_res_cache[(jobid, circ_index)] = Counter(
-                    {n: int(0.5 + d["shots"] * p) for n, p in r.items()}
-                )
+            if isinstance(res, SamplerResult):
+                # TODO Is this code still reachable?
+                for circ_index, (r, d) in enumerate(zip(res.quasi_dists, res.metadata)):
+                    self._ibm_res_cache[(jobid, circ_index)] = Counter(
+                        {n: int(0.5 + d["shots"] * p) for n, p in r.items()}
+                    )
+            else:
+                assert isinstance(res, PrimitiveResult)
+                for circ_index, pub_result in enumerate(res._pub_results):
+                    readouts = pub_result.data.c.array
```

Comment:
Is there something in the set up that allows us to assume that the register is called `c`?

Reply:
For me it is called `readout` for instance.

Reply:
A quick question: does it make sense to have multiple classical registers in a BackendResult? Or would we end up 'concatenating' them into one? If so then something like the following might prove useful?
```
counts = []
for result in res._pub_results:
    result = SamplerPubResult(result.data)
    counts.append(result.join_data().get_counts())
```
It gives a dictionary of the form `{'00': 7, '11': 3}` for instance. Hopefully that is reasonably easy to convert into a `BackendResult`? Alternatively I suppose we could use the shots directly but that would be a lot large object I think.

Reply:
Yes it does make sense, we just pack them in lexicographic order I think. Thanks for the suggestion -- I'll try it out.

Reply:
(Note that I've also ended up with a list of BackendResults, which I'm not quite sure if I should be merging or not. I think they can correspond to completely different circuits.)